src/token.rs

```diff
@@ -1,12 +1,73 @@
+use lazy_static::__Deref;
+
 /// This struct contains a token, an expiry, and an access scope.
 pub struct Token {
     // this field contains the JWT and the expiry thereof. They are in the same Option because if
     // one of them is `Some`, we require that the other be `Some` as well.
-    token: Option<(String, u64)>,
+    token: tokio::sync::RwLock<Option<(String, u64)>>,
     // store the access scope for later use if we need to refresh the token
     access_scope: String,
 }
 
+/// Trait that refreshes a token when it is expired
+#[async_trait::async_trait]
+pub trait RefreshableToken: Sync {
+    /// Getter for the token
+    async fn get_inner_token(&self) -> Option<(String, u64)>;
```

Comment:
Sorry for the nitpicking, but can we have a less verbose name here? Maybe `get_token` and `set_token`? And for many use cases, I think it would be useful to have get and set_token use `&mut self`.